src/test/regress/sql/columnar_chunk_filtering.sql

```diff
@@ -111,3 +111,106 @@ create table part_2_columnar partition of part_table for values from (0) to (150
 insert into part_table select generate_series(1,159999);
 select filtered_row_count('select count(*) from part_table where id > 75000');
 drop table part_table;
+
+-- test join parameterization
+
+set columnar.stripe_row_limit = 2000;
+set columnar.chunk_group_row_limit = 1000;
+
+create table r1(id1 int, n1 int); -- row
+create table r2(id2 int, n2 int); -- row
+create table r3(id3 int, n3 int); -- row
+
+create table coltest(id int, x1 int, x2 int, x3 int) using columnar;
+create table coltest_part(id int, x1 int, x2 int, x3 int)
+  partition by range (id);
+create table coltest_part0
+  partition of coltest_part for values from (0) to (10000)
+  using columnar;
+create table coltest_part1
+  partition of coltest_part for values from (10000) to (20000); -- row
+
+set columnar.stripe_row_limit to default;
+set columnar.chunk_group_row_limit to default;
+
+insert into r1 values(1234, 12350);
+insert into r1 values(4567, 45000);
+insert into r1 values(9101, 176000);
+insert into r1 values(14202, 7);
+insert into r1 values(18942, 189430);
+
+insert into r2 values(1234, 123502);
+insert into r2 values(4567, 450002);
+insert into r2 values(9101, 1760002);
+insert into r2 values(14202, 72);
+insert into r2 values(18942, 1894302);
+
+insert into r3 values(1234, 1235075);
+insert into r3 values(4567, 4500075);
+insert into r3 values(9101, 17600075);
+insert into r3 values(14202, 775);
+insert into r3 values(18942, 18943075);
+
+insert into coltest
+  select g, g*10, g*100, g*1000 from generate_series(0, 19999) g;
+insert into coltest_part
+  select g, g*10, g*100, g*1000 from generate_series(0, 19999) g;
+
+ANALYZE r1, r2, r3, coltest, coltest_part;
+
+-- force nested loop
+set enable_mergejoin=false;
+set enable_hashjoin=false;
+set enable_material=false;
+
+-- test different kinds of expressions
+EXPLAIN (analyze on, costs off, timing off, summary off)
+SELECT * FROM r1, coltest WHERE
+  id1 = id AND x1 > 15000 AND x1::text > '000000' AND n1 % 10 = 0;
+SELECT * FROM r1, coltest WHERE
+  id1 = id AND x1 > 15000 AND x1::text > '000000' AND n1 % 10 = 0;
```

Comment:
In general, if it wouldn't affect what we are testing, can we add `order by`s to such `SELECT` tests to make sure that they don't get flaky in future ?

Reply:
I do that sometimes, but mostly if and when it causes some headaches. Getting too much in the habit of having ORDER BY affects plans and can reduce coverage overall. Maybe in this case it's fine.